modules/cli/src/main/scala/scala/cli/exportCmd/JsonProject.scala

```diff
@@ -0,0 +1,92 @@
+package scala.cli.exportCmd
+
+import com.github.plokhotnyuk.jsoniter_scala.core.{JsonValueCodec, WriterConfig, writeToStream}
+import com.github.plokhotnyuk.jsoniter_scala.macros.{CodecMakerConfig, JsonCodecMaker}
+import coursier.Dependency
+import coursier.util.Artifact
+
+import java.nio.charset.StandardCharsets
+
+import scala.build.options.ConfigMonoid
+import scala.cli.util.SeqHelpers.*
+import scala.reflect.NameTransformer
+import scala.util.{Properties, Using}
+
+final case class JsonProject(
+  projectName: Option[String] = None,
+  scalaVersion: Option[String] = None,
+  platform: Option[String] = None,
+  scalacOptions: Seq[String] = Nil,
```

Comment:
Regarding my comment here https://github.com/VirtusLab/scala-cli/issues/1722#issuecomment-1362960092:
Why do some properties like `*Deps` or `*Sources` have a a prefix for `main` or `test` but others don't? What if e.g. a user needed an additional resolver for a dependency that is used only in tests or if they wanted to add some resources only for tests? I mean in general its possible to add using directives in a file which belongs to the test scope and they should apply to all other sources in the test scope, right? Maybe the structure here should be something like
```scala
case class JsonProject(
   projectName: Option[String],
   scalaVersion: Option[String],
   main: JsonMainProject,
   test: JsonTestProject,
   ...
)

case class JsonMainProject(
  scalacOptions: : Seq[String],
  resolvers: Seq[ExportResolverFormat],
  ...
)

case class JsonTestProject(
  scalacOptions: : Seq[String],
  resolvers: Seq[ExportResolverFormat],
  ...
)
```
???

Reply:
Good point, I'll split project information into scopes.

Reply:
I would be also good to figure out how this should behave if the project can be cross-compiled to different platforms or scala versions. Should the output json actually be a list of objects with such shape and each object would correspond to a combination of scala version and platform? Or should we return one object for the default values of these unless a user specifies them explicitly? In other words consider these two situations:
```shell
scala-cli export --json .
```

```shell
scala-cli export --json . -S 2.13 --native
```